**Windows** [Build 4896](https://ci.wildfly.org/viewLog.html?buildId=20110&buildTypeId=WF_PullRequest_Windows) outcome was **FAILURE** using a merge of 4145bbb4251a7a367760dd50976441e189c4ca15
Summary: Tests failed: 2 (1 new), passed: 3376, ignored: 198 Build time: 01:13:28
### Failed tests
```
org.jboss.as.test.manualmode.security.SecuredDataSourceTestCase.test:  at org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:146)
 at org.jboss.as.connector.subsystems.datasources.WildFlyDataSource.getConnection(WildFlyDataSource.java:66)
 at org.jboss.as.test.integration.security.common.servlets.DataSourceTestServlet.doGet(DataSourceTestServlet.java:85)
 at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
 at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
 at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:85)
 at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
 at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
 at org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
 at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
 at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:131)
 at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
 at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
 at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
 at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
 at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
 at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
 at io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
 at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
 at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
 at org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
 at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
 at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
 at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:285)
 at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:264)
 at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:81)
 at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:175)
 at io.undertow.server.Connectors.executeRootHandler(Connectors.java:202)
 at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:802)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
 at java.lang.Thread.run(Thread.java:745)
Caused by: javax.resource.ResourceException: IJ000453: Unable to get managed connection for java:jboss/datasources/SecuredDataSourceTestCase
 at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.getManagedConnection(AbstractConnectionManager.java:656)
 at org.jboss.jca.core.connectionmanager.tx.TxConnectionManagerImpl.getManagedConnection(TxConnectionManagerImpl.java:429)
 at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.allocateConnection(AbstractConnectionManager.java:747)
 at org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:138)
 ... 31 more
Caused by: javax.resource.ResourceException: IJ031084: Unable to create connection
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createLocalManagedConnection(LocalManagedConnectionFactory.java:345)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.getLocalManagedConnection(LocalManagedConnectionFactory.java:352)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.access$200(LocalManagedConnectionFactory.java:62)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory$1$1.run(LocalManagedConnectionFactory.java:259)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory$1$1.run(LocalManagedConnectionFactory.java:250)
 at java.security.AccessController.doPrivileged(Native Method)
 at javax.security.auth.Subject.doAs(Subject.java:422)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory$1.run(LocalManagedConnectionFactory.java:249)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory$1.run(LocalManagedConnectionFactory.java:246)
 at java.security.AccessController.doPrivileged(Native Method)
 at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createManagedConnection(LocalManagedConnectionFactory.java:245)
 at org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreConcurrentLinkedDequeManagedConnectionPool.createConnectionEventListener(SemaphoreConcurrentLinkedDequeManagedConnectionPool.java:1320)


org.jboss.as.test.integration.deployment.jcedeployment.JCETestCase: java.lang.RuntimeException: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.Archive org.jboss.as.test.integration.deployment.jcedeployment.JCETestCase.deployment() throws java.lang.Exception
	at java.lang.ProcessImpl.create(Native Method)
	at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
	at java.lang.ProcessImpl.start(ProcessImpl.java:137)
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
	at java.lang.Runtime.exec(Runtime.java:620)
	at java.lang.Runtime.exec(Runtime.java:450)
	at java.lang.Runtime.exec(Runtime.java:347)
	at org.jboss.as.test.integration.deployment.jcedeployment.JarSignerUtil.run(JarSignerUtil.java:61)
	at org.jboss.as.test.integration.deployment.jcedeployment.JarSignerUtil.sign(JarSignerUtil.java:85)
	at org.jboss.as.test.integration.deployment.jcedeployment.JarSignerUtil.sign(JarSignerUtil.java:71)
	at org.jboss.as.test.integration.deployment.jcedeployment.JCETestCase.deployment(JCETestCase.java:74)
------- Stdout: -------
16:17:08,080 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-5) WFLYSRV0028: Stopped deployment excludeSubsystem.jar (runtime-name: excludeSubsystem.jar) in 5ms
16:17:08,092 WARN  [org.jboss.as.controller] (management-handler-thread - 2) WFLYCTL0357: Notification of type deployment-undeployed is not described for the resource at the address []
16:17:08,094 INFO  [org.jboss.as.repository] (management-handler-thread - 2) WFLYDR0002: Content removed from location C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\basic\target\jbossas\standalone\data\content\26\b29da36a1a95a9206b4a2666884fb026aecc6f\content
16:17:08,094 INFO  [org.jboss.as.server] (management-handler-thread - 2) WFLYSRV0009: Undeployed "excludeSubsystem.jar" (runtime-name: "excludeSubsystem.jar")


```

--- a/pom.xml
+++ b/pom.xml
@@ -125,8 +125,8 @@
         <version.org.apache.james.apache-mime4j>0.6</version.org.apache.james.apache-mime4j>
         <version.org.apache.lucene>5.3.1</version.org.apache.lucene>
         <version.org.apache.myfaces.core>2.1.17</version.org.apache.myfaces.core>
-        <version.org.apache.openjpa>2.4.0</version.org.apache.openjpa>
-        <version.org.eclipselink.version>2.6.0</version.org.eclipselink.version>
+        <version.org.apache.openjpa>2.4.1</version.org.apache.openjpa>
+        <version.org.eclipselink.version>2.6.3</version.org.eclipselink.version>
         <version.org.apache.neethi>3.0.3</version.org.apache.neethi>
         <version.org.apache.qpid.proton>0.8</version.org.apache.qpid.proton>
         <version.org.apache.santuario>2.0.6</version.org.apache.santuario>
@@ -138,7 +138,7 @@
         <version.org.codehaus.jackson>1.9.13</version.org.codehaus.jackson>
         <version.org.codehaus.jettison>1.3.3</version.org.codehaus.jettison>
         <version.org.cryptacular>1.0</version.org.cryptacular>
-        <version.org.eclipse.jdt.core.compiler>4.4.2</version.org.eclipse.jdt.core.compiler>
+        <version.org.eclipse.jdt.core.compiler>4.5.1</version.org.eclipse.jdt.core.compiler>
         <version.org.glassfish.javax.el>3.0.1-b08-jbossorg-1</version.org.glassfish.javax.el>
         <version.org.glassfish.javax.enterprise.concurrent>1.0</version.org.glassfish.javax.enterprise.concurrent>
         <version.org.glassfish.javax.json>1.0.3</version.org.glassfish.javax.json>

--- a/pom.xml
+++ b/pom.xml
@@ -6127,6 +6127,8 @@
                 </modular.jdk.args>
                 <!-- [WFCORE-1494] remove JUL workaround -->
                 <modular.jdk.props>-Dsun.util.logging.disableCallerCheck=true</modular.jdk.props>
+                <!-- There are lots of issues with checkstyle runtime on JDK9, it somewhat works but really slows down build, disabling it for now-->
+                <checkstyle.skip>true</checkstyle.skip>
             </properties>
             <build>
                 <plugins>

--- a/testsuite/compat/pom.xml
+++ b/testsuite/compat/pom.xml
@@ -50,6 +50,7 @@
         <!-- Used to provide an absolute location for the XSLT scripts. -->
         <!-- This value is overridden in submodules with the correct relative path. -->
         <xslt.scripts.dir>${basedir}/../integration/src/test/xslt</xslt.scripts.dir>
+        <surefire.excludeTests />
     </properties>
 
     <!--
@@ -82,12 +83,6 @@
             <scope>test</scope>
         </dependency>
 
-        <dependency>
-            <groupId>net.sourceforge.serp</groupId>
-            <artifactId>serp</artifactId>
-            <version>1.14.1</version>
-            <scope>test</scope>
-        </dependency>
         <dependency>
             <groupId>org.jboss.arquillian.junit</groupId>
             <artifactId>arquillian-junit-container</artifactId>
@@ -159,20 +154,6 @@
             </exclusions>
             </dependency>
 
-            <dependency>
-                <groupId>org.apache.openjpa</groupId>
-                <artifactId>openjpa-all</artifactId>
-                <version>2.2.2</version>
-                <scope>test</scope>
-            </dependency>
-
-            <dependency>
-                <groupId>net.sourceforge.serp</groupId>
-                <artifactId>serp</artifactId>
-                <version>1.14.1</version>
-                <scope>test</scope>
-            </dependency>
-
         </dependencies>
     </dependencyManagement>
 
@@ -217,7 +198,9 @@
 
                     <!-- System properties to forked surefire JVM which runs clients. -->
                     <argLine>${jvm.args.ip.client} ${jvm.args.timeouts}</argLine>
-
+                    <excludes>
+                        <exclude>${surefire.excludeTests}</exclude>
+                    </excludes>
                     <!-- System properties passed to test cases -->
                     <systemPropertyVariables combine.children="append">
                         <node0>${node0}</node0>
@@ -244,52 +227,16 @@
 
     <profiles>
         <profile>
-            <id>default</id>
+            <id>jdk9</id>
             <activation>
-                <activeByDefault>true</activeByDefault>
+                <jdk>9</jdk>
             </activation>
-            <dependencies>
-                <dependency>
-                    <groupId>org.wildfly.arquillian</groupId>
-                    <artifactId>wildfly-arquillian-container-managed</artifactId>
-                    <scope>test</scope>
-                </dependency>
-            </dependencies>
-        </profile>
-
-        <profile>
-            <id>remote</id>
-            <activation>
-                <property>
-                    <name>remote</name>
-                </property>
-            </activation>
-            <dependencies>
-                <dependency>
-                    <groupId>org.wildfly.arquillian</groupId>
-                    <artifactId>wildfly-arquillian-container-remote</artifactId>
-                    <scope>test</scope>
-                </dependency>
-            </dependencies>
+            <properties>
+                <!--WFLY-6777 We disable OpenJPA test on JDK9, as openjdk-s serp bytecode enhancer hangs deployment. -->
+                <surefire.excludeTests>**/OpenJPASharedModuleProviderTestCase.java</surefire.excludeTests>
+            </properties>
         </profile>
     </profiles>
 
-  <repositories>
-
-    <repository>
-      <id>jboss-public-repository-group</id>
-      <name>JBoss Public Repository Group</name>
-      <url>http://repository.jboss.org/nexus/content/groups/public/</url>
-      <layout>default</layout>
-      <releases>
-        <enabled>true</enabled>
-        <updatePolicy>never</updatePolicy>
-      </releases>
-      <snapshots>
-        <enabled>true</enabled>
-        <updatePolicy>never</updatePolicy>
-      </snapshots>
-    </repository>
-  </repositories>
 
 </project>

--- a/testsuite/compat/src/test/java/org/jboss/as/test/compat/jpa/openjpa/OpenJPASharedModuleProviderTestCase.java
+++ b/testsuite/compat/src/test/java/org/jboss/as/test/compat/jpa/openjpa/OpenJPASharedModuleProviderTestCase.java
@@ -56,7 +56,8 @@ public class OpenJPASharedModuleProviderTestCase {
         "       <properties>" +
         "           <property name=\"openjpa.jdbc.SynchronizeMappings\" value=\"buildSchema(SchemaAction='drop,add')\"/>" +
         "           <property name=\"openjpa.InitializeEagerly\" value=\"true\"/>" +
-        "           <property name=\"openjpa.RuntimeUnenhancedClasses\" value=\"supported\"/>" +
+        "           <property name=\"openjpa.RuntimeUnenhancedClasses\" value=\"unsupported\"/>" +
+        "           <property name=\"openjpa.DynamicEnhancementAgent\" value=\"false\"/>" +
         "           <property name=\"jboss.as.jpa.providerModule\" value=\"org.apache.openjpa:test\"/>" +
         "       </properties>" +
         "  </persistence-unit>" +

--- a/testsuite/compat/src/test/scripts/build-jars.xml
+++ b/testsuite/compat/src/test/scripts/build-jars.xml
@@ -71,8 +71,6 @@
     <!-- create the openjpa module to test with -->
     <copy file="${org.apache.openjpa:openjpa-all:jar}"         tofile="${tests.output.dir}/modules/org/apache/openjpa/test/openjpa-all.jar"/>
     <copy file="${org.wildfly:jipijapa-openjpa:jar}"      tofile="${tests.output.dir}/modules/org/apache/openjpa/test/jipijapa-openjpa.jar"/>
-    <copy file="${net.sourceforge.serp:serp:jar}"      tofile="${tests.output.dir}/modules/org/apache/openjpa/test/serp.jar"/>
-
     <echo file="${tests.output.dir}/modules/org/apache/openjpa/test/module.xml" append="false">&lt;?xml version="1.0" encoding="UTF-8"?&gt;
 &lt;module xmlns="urn:jboss:module:1.1" name="org.apache.openjpa" slot="test"&gt;
 &lt;resources&gt;
@@ -84,7 +82,6 @@
 &lt;/resource-root&gt;
 
 &lt;resource-root path="jipijapa-openjpa.jar"/&gt;
-&lt;resource-root path="serp.jar"/&gt;
 &lt;/resources&gt;
 
 &lt;dependencies&gt;

--- a/testsuite/domain/pom.xml
+++ b/testsuite/domain/pom.xml
@@ -178,6 +178,10 @@
             <groupId>com.io7m.xom</groupId>
             <artifactId>xom</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.codehaus.woodstox</groupId>
+            <artifactId>woodstox-core-asl</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/testsuite/domain/pom.xml
+++ b/testsuite/domain/pom.xml
@@ -229,7 +229,8 @@
                     <argLine>${jvm.args.ip.client} ${jvm.args.timeouts}</argLine>
 
                     <systemPropertyVariables>
-                        <jboss.options>${surefire.system.args}</jboss.options>
+                        <!-- Disabled assertions until new version of core with fix for WFCORE-1630 is used -->
+                        <!--<jboss.options>-ea</jboss.options>-->
                         <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                         <jboss.home>${project.basedir}/target/jbossas</jboss.home>
                         <module.path>${jboss.dist}/modules</module.path>
@@ -286,8 +287,8 @@
                             <argLine>${jvm.args.ip.client} ${jvm.args.timeouts}</argLine>
 
                             <systemPropertyVariables>
-                                <jboss.options>${surefire.system.args}</jboss.options>
                                 <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                                <jboss.options>-ea</jboss.options>
                                 <jboss.home>${project.basedir}/target/jbossas</jboss.home>
                                 <module.path>${jboss.dist}/modules</module.path>
                                 <jboss.test.host.master.address>${node0}</jboss.test.host.master.address>
@@ -299,6 +300,7 @@
                             <includes>
                                 <include>org/jboss/as/test/integration/domain/rbac/RbacSoakTest.java</include>
                             </includes>
+                            <argLine />
                         </configuration>
                     </plugin>
                 </plugins>

--- a/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java
+++ b/testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java
@@ -64,10 +64,15 @@ public class MixedDomainTestSupport extends DomainTestSupport {
     private MixedDomainTestSupport(Version.AsVersion version, String testClass, String domainConfig, String masterConfig, String slaveConfig,
                                    String jbossHome, boolean adjustDomain, boolean legacyConfig)
             throws Exception {
-        super(testClass, domainConfig, masterConfig, slaveConfig, new WildFlyManagedConfiguration(), new WildFlyManagedConfiguration(jbossHome));
+        super(testClass, domainConfig, masterConfig, slaveConfig, configWithDisabledAsserts(null), configWithDisabledAsserts(jbossHome));
         this.version = version;
         this.adjustDomain = adjustDomain;
         this.legacyConfig = legacyConfig;
+    }
+    private static WildFlyManagedConfiguration configWithDisabledAsserts(String jbossHome){
+        WildFlyManagedConfiguration config = new WildFlyManagedConfiguration(jbossHome);
+        config.setEnableAssertions(false);
+        return config;
     }
 
     public static MixedDomainTestSupport create(String testClass, Version.AsVersion version) throws Exception {

--- a/testsuite/pom.xml
+++ b/testsuite/pom.xml
@@ -145,7 +145,7 @@
         <surefire.jpda.args></surefire.jpda.args>
         <as.debug.port>8787</as.debug.port>
         <modular.jdk.testsuite.args />
-        <surefire.system.args> ${modular.jdk.testsuite.args} -Dorg.jboss.ejb.client.wildfly-testsuite-hack=true ${surefire.memory.args} ${surefire.jpda.args} -Djboss.dist=${jboss.dist}</surefire.system.args>
+        <surefire.system.args>${modular.jdk.testsuite.args} -Dorg.jboss.ejb.client.wildfly-testsuite-hack=true ${surefire.memory.args} ${surefire.jpda.args} -Djboss.dist=${jboss.dist}</surefire.system.args>
 
         <!-- If servers should be killed before the test suite is run-->
         <org.wildfly.test.kill-servers-before-test>false</org.wildfly.test.kill-servers-before-test>
@@ -329,7 +329,6 @@
                        <failIfNoTests>false</failIfNoTests>
                        <redirectTestOutputToFile>${testLogToFile}</redirectTestOutputToFile>
                        <systemPropertyVariables>
-                           <jboss.options>${surefire.system.args} ${jvm.args.jacoco}</jboss.options>
                            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
                            <jboss.dist>${jboss.dist}</jboss.dist>
                            <jboss.home>${basedir}/target/jbossas</jboss.home>

